From: Patrick Brunschwig <patrick@enigmail.net>
Date: Mon, 25 Jan 2016 07:46:36 +0100
Subject: fixed typo

---
 ui/content/enigmailWrapSelection.xul | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/content/enigmailWrapSelection.xul b/ui/content/enigmailWrapSelection.xul
index 88c0dcf..2e79c4d 100644
--- a/ui/content/enigmailWrapSelection.xul
+++ b/ui/content/enigmailWrapSelection.xul
@@ -31,7 +31,7 @@
     <description>&enigmail.compose.excesslength_action.label;</description>
     <vbox>
       <groupbox>
-        <radiogroup id="WrapSelectGroup" aria-labelledby="enigmailWrapDesc">
+        <radiogroup id="wrapSelectGroup" aria-labelledby="enigmailWrapDesc">
           <radio value="0" id="wrap"
               label="&enigmail.compose.wrap_radio.label;"/>
           <radio value="1" id="sendasis"
