From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Tue, 7 Jul 2015 15:43:28 -0400
Subject: stabilize the order of the keys in generated .dtds

---
 util/fixlang.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/fixlang.pl b/util/fixlang.pl
index db3e94b..7a92758 100755
--- a/util/fixlang.pl
+++ b/util/fixlang.pl
@@ -128,7 +128,7 @@ my $frdtd = loaddtd("$dest/enigmail.dtd");
 print "+ Writing $dest/enigmail.dtd\n";
 open(OUT, ">$dest/enigmail.dtd.gen")  || die "Cannot write to $dest/enigmail.dtd";
 
-for my $ind (keys %$endtd) {
+for my $ind (sort keys %$endtd) {
 
   if ($frdtd->{$ind}) {
     print OUT "<!ENTITY $frdtd->{$ind}\n";
@@ -146,7 +146,7 @@ my $frprop = loadprop("$dest/enigmail.properties");
 
 print "+ Writing $dest/enigmail.properties\n";
 open(OUT, ">$dest/enigmail.properties.gen") || die "Cannot write to $dest/enigmail.properties";
-for my $ind (keys %$enprop) {
+for my $ind (sort keys %$enprop) {
   if ($frprop->{$ind}) {
     print OUT "$frprop->{$ind}\n";
   } else {
