From: Patrick Brunschwig <patrick@enigmail.net>
Date: Thu, 8 Sep 2016 08:10:34 +0200
Subject: fix for pgpMime forwarding issue

---
 package/mimeDecrypt.jsm | 1 -
 1 file changed, 1 deletion(-)

diff --git a/package/mimeDecrypt.jsm b/package/mimeDecrypt.jsm
index 180d7ab..4b70e5f 100644
--- a/package/mimeDecrypt.jsm
+++ b/package/mimeDecrypt.jsm
@@ -449,7 +449,6 @@ EnigmailMimeDecrypt.prototype = {
         veri.onStopRequest(null, null, 0);
       }
       else {
-        gConv.setData(data, data.length);
         this.mimeSvc.onStartRequest(null, null);
         this.mimeSvc.onDataAvailable(null, null, gConv, 0, data.length);
         this.mimeSvc.onStopRequest(null, null, 0);
