From: Alexander Sack <asac@debian.org>
Date: Sun, 25 Mar 2012 17:40:03 +0200
Subject: fix-xpidl-path

===================================================================
---
 config/config.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/config.mk b/config/config.mk
index a830bfe..3ae203f 100644
--- a/config/config.mk
+++ b/config/config.mk
@@ -408,7 +408,7 @@ QUANTIFY = quantify $(QUANTIFYOPTIONS)
 ifdef CROSS_COMPILE
 XPIDL_COMPILE = $(LIBXUL_DIST)/host/bin/host_xpidl$(HOST_BIN_SUFFIX)
 else
-XPIDL_COMPILE = $(LIBXUL_DIST)/bin/xpidl$(BIN_SUFFIX)
+XPIDL_COMPILE = $(LIBXUL_DIST)/sdk/bin/xpidl$(BIN_SUFFIX)
 endif
 XPIDL_LINK = $(PYTHON) $(LIBXUL_DIST)/sdk/bin/xpt.py link
 
