From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 6 Feb 2013 19:19:46 +0100
Subject: default-to-PGP-MIME
Bug: http://bugs.debian.org/698084

---
 package/prefs/enigmail.js |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/package/prefs/enigmail.js
+++ b/package/prefs/enigmail.js
@@ -238,7 +238,7 @@
 pref("mail.identity.default.defaultEncryptionPolicy", 0);
 pref("mail.identity.default.openPgpHeaderMode", 0);
 pref("mail.identity.default.openPgpUrlName", "");
-pref("mail.identity.default.pgpMimeMode", false);
+pref("mail.identity.default.pgpMimeMode", true);
 pref("mail.identity.default.attachPgpKey", false);
 pref("mail.identity.default.autoEncryptDrafts", true);
 
