From: Alexander Sack <asac@debian.org>
Date: Sun, 25 Mar 2012 17:40:03 +0200
Subject: dont_register_cids_multiple_times

---
 extensions/enigmail/build/nsEnigModule.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/extensions/enigmail/build/nsEnigModule.cpp b/extensions/enigmail/build/nsEnigModule.cpp
index d2baf06..fd05790 100644
--- a/extensions/enigmail/build/nsEnigModule.cpp
+++ b/extensions/enigmail/build/nsEnigModule.cpp
@@ -78,14 +78,14 @@ const mozilla::Module::CIDEntry kEnigModuleCIDs[] = {
   { &kNS_PIPEFILTERLISTENER_CID, false, NULL, nsPipeFilterListenerConstructor },
   { &kNS_ENIGMSGCOMPOSE_CID, false, NULL, nsEnigMsgComposeConstructor },
   { &kNS_ENIGMSGCOMPOSEFACTORY_CID, false, NULL, nsEnigMsgComposeFactoryConstructor },
-  { &kNS_ENIGMSGCOMPOSE_CID, false, NULL, nsEnigMsgComposeFactoryConstructor },
+//  { &kNS_ENIGMSGCOMPOSE_CID, false, NULL, nsEnigMsgComposeFactoryConstructor },
   { &kNS_ENIGMIMELISTENER_CID, false, NULL, nsEnigMimeListenerConstructor },
   { &kNS_ENIGMIMEWRITER_CID, false, NULL, nsEnigMimeWriterConstructor },
   { &kNS_ENIGMIMEDECRYPT_CID, false, NULL, nsEnigMimeDecryptConstructor },
   { &kNS_ENIGMIMEVERIFY_CID, false, NULL, nsEnigMimeVerifyConstructor },
   { &kNS_ENIGMIMESERVICE_CID, false, NULL, nsEnigMimeServiceConstructor },
   { &kNS_ENIGCONTENTHANDLER_CID, false, NULL, nsEnigContentHandlerConstructor },
-  { &kNS_ENIGCONTENTHANDLER_CID, false, NULL, nsEnigContentHandlerConstructor },
+//  { &kNS_ENIGCONTENTHANDLER_CID, false, NULL, nsEnigContentHandlerConstructor },
   { NULL }
 };
 
