From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 30 May 2018 15:31:50 -0400
Subject: avoid OpenPGP.js when building

---
 stdlib/Makefile | 1 -
 util/genxpi     | 3 ---
 2 files changed, 4 deletions(-)

diff --git a/stdlib/Makefile b/stdlib/Makefile
index 09841e3..2234775 100644
--- a/stdlib/Makefile
+++ b/stdlib/Makefile
@@ -13,7 +13,6 @@ MODFILES = \
            compose.jsm \
            misc.jsm \
            msgHdrUtils.jsm \
-           openpgp.worker.js \
            web-streams.jsm
 
 GENFILES = $(addprefix $(GENDIR)/,$(MODFILES))
diff --git a/util/genxpi b/util/genxpi
index 9d7c39e..68d0b81 100755
--- a/util/genxpi
+++ b/util/genxpi
@@ -81,7 +81,6 @@ zip -9 --must-match\
     chrome/content/modules/cryptoAPI/gnupg-decryption.jsm \
     chrome/content/modules/cryptoAPI/gnupg-key.jsm \
     chrome/content/modules/cryptoAPI/gnupg-keylist.jsm \
-    chrome/content/modules/cryptoAPI/openpgp-js.js \
     chrome/content/modules/cryptoAPI/interface.js \
     chrome/content/modules/data.jsm \
     chrome/content/modules/dns.jsm \
@@ -149,8 +148,6 @@ zip -9 --must-match\
     chrome/content/modules/stdlib/compose.jsm \
     chrome/content/modules/stdlib/misc.jsm \
     chrome/content/modules/stdlib/msgHdrUtils.jsm \
-    chrome/content/modules/stdlib/openpgp-lib.jsm \
-    chrome/content/modules/stdlib/openpgp.worker.js \
     chrome/content/modules/stdlib/web-streams.jsm \
     chrome/content/modules/stdlib.jsm \
     chrome/content/modules/streams.jsm \
