From: Willi Mann <willi@debian.org>
Date: Sat, 15 Dec 2012 08:43:10 +0100
Subject: install.rdf: Require at least icedove 17

---
 extensions/enigmail/package/install.rdf |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extensions/enigmail/package/install.rdf b/extensions/enigmail/package/install.rdf
index 3279a41..f075db6 100644
--- a/extensions/enigmail/package/install.rdf
+++ b/extensions/enigmail/package/install.rdf
@@ -15,7 +15,7 @@
       <!-- Thunderbird -->
       <Description>
         <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
-        <em:minVersion>16.0</em:minVersion>
+        <em:minVersion>17.0</em:minVersion>
         <em:maxVersion>19.0pre</em:maxVersion>
       </Description>
     </em:targetApplication>
