--- a/config/getOsTarget.pl
+++ b/config/getOsTarget.pl
@@ -79,6 +79,9 @@
 elsif ($srcCpu =~ /ppc/i) {
   $targetCpu = "ppc";
 }
+elsif ($srcCpu =~ /powerpc/i) {
+  $targetCpu = "ppc";
+}
 elsif ($srcCpu =~ /alpha/i) {
   $targetCpu = "Alpha";
 }
@@ -89,7 +92,10 @@
   $targetCpu = "ia64";
 }
 elsif ($srcCpu =~ /arm/i) {
-  $targetCpu = "arm";
+  $targetCpu = "arm-eabi";
+}
+elsif ($srcCpu =~ /mips/i) {
+  $targetCpu = "mips";
 }
 else {
   $targetCpu = $srcCpu;
