From: Patrick Brunschwig <patrick@enigmail.net>
Date: Sun, 12 Jun 2016 14:22:42 +0200
Subject: suppress message if key import was successful

---
 ui/content/enigmailSearchKey.js | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/ui/content/enigmailSearchKey.js b/ui/content/enigmailSearchKey.js
index 652c3cf..96505bb 100644
--- a/ui/content/enigmailSearchKey.js
+++ b/ui/content/enigmailSearchKey.js
@@ -377,13 +377,12 @@ function importKeybaseKeys(txt) {
           gEnigRequest.dlKeyList[gEnigRequest.keyNum - 1],
           errorMsgObj);
 
-        if (errorMsgObj.value) {
-          EnigmailDialog.alert(window, errorMsgObj.value);
-        }
-
         if (r === 0) {
           window.arguments[RESULT].importedKeys++;
         }
+        else if (errorMsgObj.value) {
+          EnigmailDialog.alert(window, errorMsgObj.value);
+        }
       }
     }
   }
