From: Patrick Brunschwig <patrick@enigmail.net>
Date: Sun, 27 Mar 2016 18:44:02 +0200
Subject: handle situation with EnigmailVerify.lastMsgWindow being null

---
 package/pgpmimeHandler.js | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/package/pgpmimeHandler.js b/package/pgpmimeHandler.js
index 1e7b04e..be5c378 100644
--- a/package/pgpmimeHandler.js
+++ b/package/pgpmimeHandler.js
@@ -88,14 +88,38 @@ PgpMimeHandler.prototype = {
   onStopRequest: function(request, win, status) {},
 
   handleSmime: function(uri) {
+    let headerSink;
     this.contentHandler = throwErrors;
 
     if (uri) {
       uri = uri.QueryInterface(Ci.nsIURI).clone();
     }
-    let headerSink = EnigmailVerify.lastMsgWindow.msgHeaderSink.securityInfo.QueryInterface(Ci.nsIEnigMimeHeaderSink);
+
+    if (EnigmailVerify.lastMsgWindow) {
+      headerSink = EnigmailVerify.lastMsgWindow.msgHeaderSink.securityInfo.QueryInterface(Ci.nsIEnigMimeHeaderSink);
+    }
+    else {
+      let domWin = this.getMessengerWindow();
+      headerSink = domWin.msgWindow.msgHeaderSink.securityInfo.QueryInterface(Ci.nsIEnigMimeHeaderSink);
+    }
+
     headerSink.handleSMimeMessage(uri);
-  }
+  },
+
+  getMessengerWindow: function() {
+    let windowManager = Cc["@mozilla.org/appshell/window-mediator;1"].getService(Ci.nsIWindowMediator);
+
+    let winEnum = windowManager.getEnumerator(null);
+
+    while (winEnum.hasMoreElements()) {
+      let thisWin = winEnum.getNext();
+      if (thisWin.location.href.search(/\/messenger.xul$/) > 0) {
+        return thisWin;
+      }
+    }
+
+    return null;
+  },
 };
 
 
