From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Thu, 6 Sep 2018 02:00:31 -0400
Subject: add more logging to autocrypt test

---
 package/tests/autocrypt-test.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/package/tests/autocrypt-test.js b/package/tests/autocrypt-test.js
index dd91642..9582531 100644
--- a/package/tests/autocrypt-test.js
+++ b/package/tests/autocrypt-test.js
@@ -8,7 +8,7 @@
 "use strict";
 
 do_load_module("file://" + do_get_cwd().path + "/testHelper.js");
-/* global setupTestAccounts: false, withTestGpgHome: false, withEnigmail: false, component: false */
+/* global setupTestAccounts: false, withTestGpgHome: false, withEnigmail: false, component: false, withLogFiles: false */
 
 testing("autocrypt.jsm"); /*global EnigmailAutocrypt: false, EnigmailKeyRing: false, EnigmailStdlib: false, EnigmailSqliteDb: false */
 const Sqlite = ChromeUtils.import("resource://gre/modules/Sqlite.jsm").Sqlite;
@@ -170,7 +170,7 @@ test(function processHeader() {
 
 });
 
-test(withTestGpgHome(withEnigmail(function shouldGetKeyFunctions() {
+test(withLogFiles(withTestGpgHome(withEnigmail(function shouldGetKeyFunctions() {
   const publicKey = do_get_file("resources/dev-strike.asc", false);
   const secretKey = do_get_file("resources/dev-strike.sec", false);
   EnigmailKeyRing.importKeyFromFile(publicKey, {}, {});
@@ -199,4 +199,4 @@ test(withTestGpgHome(withEnigmail(function shouldGetKeyFunctions() {
   });
   inspector.enterNestedEventLoop(0);
 
-})));
\ No newline at end of file
+}))));
