From: Patrick Brunschwig <patrick@enigmail.net>
Date: Mon, 28 Mar 2016 20:10:11 +0200
Subject: fixed CSS syntax errors

---
 ui/content/keyDetailsDlg.xul | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui/content/keyDetailsDlg.xul b/ui/content/keyDetailsDlg.xul
index 3ed1a0e..d315f4b 100644
--- a/ui/content/keyDetailsDlg.xul
+++ b/ui/content/keyDetailsDlg.xul
@@ -66,7 +66,7 @@
     </vbox>
     <vbox flex="1" align="end">
       <hbox flex="1" align="end">
-        <image src="" id="photoIdImg" style="display: block, height: auto; width: auto, max-width: 100px; max-height: 120px;"/>
+        <image src="" id="photoIdImg" style="display: block; height: auto; width: auto; max-width: 100px; max-height: 120px;"/>
       </hbox>
     </vbox>
   </hbox>
