diff --git a/extensions/enigmail/package/enigmail.js b/extensions/enigmail/package/enigmail.js
index 69ec5c3..076906a 100644
--- a/extensions/enigmail/package/enigmail.js
+++ b/extensions/enigmail/package/enigmail.js
@@ -1752,7 +1752,13 @@ Enigmail.prototype = {
     // Wait for child to close
     proc.wait();
 
-    var plainText = listener.stdoutData;
+    // convert the output to Unicode
+    var tmpStream = Cc["@mozilla.org/io/string-input-stream;1"].createInstance(Ci.nsIStringInputStream);
+    tmpStream.setData(listener.stdoutData, listener.stdoutData.length);
+    var inStream = Cc["@mozilla.org/scriptableinputstream;1"].createInstance(Ci.nsIScriptableInputStream);
+    inStream.init(tmpStream);
+    var plainText = inStream.read(tmpStream.available());
+
     Ec.DEBUG_LOG("enigmail.js: decryptMessage: got plaintext: '"+plainText+"'\n");
 
     var retStatusObj = {};
