From: Willi Mann <willi@wm1.at>
Date: Wed, 9 May 2012 21:59:29 +0200
Subject: Require Icedove 11 for 1.4.1exp

---
 extensions/enigmail/package/install.rdf |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extensions/enigmail/package/install.rdf b/extensions/enigmail/package/install.rdf
index becdd1d..08f4ea5 100644
--- a/extensions/enigmail/package/install.rdf
+++ b/extensions/enigmail/package/install.rdf
@@ -15,7 +15,7 @@
       <!-- Thunderbird -->
       <Description>
         <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
-        <em:minVersion>10.0</em:minVersion>
+        <em:minVersion>11.0</em:minVersion>
         <em:maxVersion>14.0pre</em:maxVersion>
       </Description>
     </em:targetApplication>
