Author: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: 2013-11-08 21:20:42-0500

This is sheer pedantry, but it's C code, not C++ :)

--- a/ipc/src/subprocess.c
+++ b/ipc/src/subprocess.c
@@ -8,12 +8,12 @@
 
 void closeOtherFds(int fdIn, int fdOut, int fdErr) {
 
-  int maxFD = 256; // arbitrary max
+  int maxFD = 256; /* arbitrary max */
   int i;
   struct rlimit rl;
 
   if (getrlimit(RLIMIT_NOFILE, &rl) == 0) {
-      if (rl.rlim_cur <  999999) // ignore too high numbers
+      if (rl.rlim_cur <  999999) /* ignore too high numbers */
         maxFD = rl.rlim_cur;
   }
 
@@ -37,7 +37,7 @@
 
   pid = fork();
   if (pid == 0) {
-    // child
+    /* child */
     if (workdir) {
       if (chdir(workdir) < 0) {
         _exit(126);
@@ -60,6 +60,6 @@
     _exit(1);
   }
 
-  // parent
+  /* parent */
   return pid;
 }
