From: Patrick Brunschwig <patrick@enigmail.net>
Date: Sat, 25 Jun 2016 19:12:15 +0200
Subject: improved previous patch

---
 package/key.jsm | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/package/key.jsm b/package/key.jsm
index 274bb91..f90022c 100644
--- a/package/key.jsm
+++ b/package/key.jsm
@@ -279,17 +279,19 @@ var EnigmailKey = {
     var keyexpired;
 
     while (state != "end") {
-      if (idx >= lines.length) {
-        errorMsgObj.value = EnigmailLocale.getString("cantImport");
-        return [];
-      }
 
       // Ignore all irrelevant lines
-      while (lines[idx].search(/^(IMPORTED|IMPORT_OK|IMPORT_RES|IMPORT_PROBLEM) /) < 0) {
+      while (lines[idx].search(/^(IMPORTED|IMPORT_OK|IMPORT_RES|IMPORT_PROBLEM) /) < 0 &&
+        idx < lines.length) {
         EnigmailLog.DEBUG("Ignoring line: '" + lines[idx] + "'\n");
         ++idx;
       }
 
+      if (idx >= lines.length) {
+        errorMsgObj.value = EnigmailLocale.getString("cantImport");
+        return [];
+      }
+
       EnigmailLog.DEBUG("state: '" + state + "', line: '" + lines[idx] + "'\n");
 
       switch (state) {
