From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Tue, 5 Nov 2019 21:12:37 -0500
Subject: avoid trying to test openpgpjs

---
 package/tests/main.js | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/package/tests/main.js b/package/tests/main.js
index fd68452..9b9397f 100644
--- a/package/tests/main.js
+++ b/package/tests/main.js
@@ -57,6 +57,5 @@ execTest("rng-test.js");
 execTest("dns-test.js");
 execTest("filters-test.js");
 execTest("webKey-test.js");
-execTest("openpgpjs-test.js");
 execTest("autoSetup-test.js");
-execTest("persistentCrypto-test.js");
\ No newline at end of file
+execTest("persistentCrypto-test.js");
