From: Patrick Brunschwig <patrick@enigmail.net>
Date: Mon, 31 Oct 2016 08:11:47 +0100
Subject: ensure that the temp directory is unique

---
 package/key.jsm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/key.jsm b/package/key.jsm
index ab4b739..4f10a1d 100644
--- a/package/key.jsm
+++ b/package/key.jsm
@@ -253,7 +253,7 @@ var EnigmailKey = {
       return ret;
     }
 
-    const tempDir = EnigmailFiles.createTempSubDir("enigmail_import");
+    const tempDir = EnigmailFiles.createTempSubDir("enigmail_import", true);
     const tempPath = EnigmailFiles.getFilePath(tempDir);
     const args = EnigmailGpg.getStandardArgs(true).concat([
       "--import",
