From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Sun, 24 Jun 2018 19:57:56 -0400
Subject: write debug message with clearer error message

---
 package/autocrypt.jsm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/autocrypt.jsm b/package/autocrypt.jsm
index 048903f..1eb6cdc 100644
--- a/package/autocrypt.jsm
+++ b/package/autocrypt.jsm
@@ -300,7 +300,7 @@ var EnigmailAutocrypt = {
           return checkDatabaseStructure(conn);
         },
         function onError(error) {
-          EnigmailLog.DEBUG("autocrypt.jsm: getOpenPGPKeyForEmail: could not open database\n");
+          EnigmailLog.DEBUG("autocrypt.jsm: getOpenPGPKeyForEmail: could not open database [" + error + "]\n");
           reject("getOpenPGPKeyForEmail1 error " + error);
           return;
         }
