Index: enigmail-1.2.1/extensions/enigmail/genxpi
===================================================================
--- enigmail-1.2.1.orig/extensions/enigmail/genxpi	2011-08-12 10:19:40.000000000 +0200
+++ enigmail-1.2.1/extensions/enigmail/genxpi	2011-08-12 10:19:42.000000000 +0200
@@ -68,7 +68,7 @@
 
 
 enigmimeDll=${libPrefix}enigmime${dllSuffix}
-ipcDll=${libPrefix}ipc${dllSuffix}
+ipcDll=${libPrefix}ipc-pipe${dllSuffix}
 if [ ${platform} = OS2_x86-gcc3 ]; then
   # workaround for file name length limit on OS/2
   enigDllFile=platform/${platform}/components/${enigmimeDll}
@@ -185,7 +185,7 @@
     wrappers/gpg-wrapper.sh \
     wrappers/gpg-agent-wrapper.sh \
     ${localeJar} \
-    components/ipc.xpt \
+    components/ipc-pipe.xpt \
     components/enigmime.xpt \
     ${enigDllFile} \
     ${ipcDllFile} \
Index: enigmail-1.2.1/extensions/enigmail/package/chrome.manifest
===================================================================
--- enigmail-1.2.1.orig/extensions/enigmail/package/chrome.manifest	2011-08-12 10:19:40.000000000 +0200
+++ enigmail-1.2.1/extensions/enigmail/package/chrome.manifest	2011-08-12 10:19:42.000000000 +0200
@@ -65,7 +65,7 @@
 resource enigmail modules/
 
 ## Gecko 2.0 interface specifiacations
-interfaces  components/ipc.xpt
+interfaces  components/ipc-pipe.xpt
 interfaces  components/enigmime.xpt
 interfaces  components/enigmail.xpt
 
