Index: enigmail-1.0.1/extensions/enigmail/Makefile.in
===================================================================
--- enigmail-1.0.1.orig/extensions/enigmail/Makefile.in	2010-06-01 17:38:45.000000000 +0200
+++ enigmail-1.0.1/extensions/enigmail/Makefile.in	2010-06-01 17:39:39.000000000 +0200
@@ -95,7 +95,7 @@
 endif
 
 xpi::
-	$(srcdir)/genxpi $(XPIFILE) $(XPI_MODULE_VERS) $(OS_ARCH) $(TARGET_XPCOM_ABI) $(MOZ_APP_NAME)-$(MOZ_APP_VERSION) $(DIST)/bin $(srcdir) $(XPI_MODULE) $(DLL_SUFFIX) $(LIB_PREFIX)
+	$(srcdir)/genxpi $(XPIFILE) $(XPI_MODULE_VERS) $(OS_ARCH) "$(TARGET_XPCOM_ABI)" $(MOZ_APP_NAME)-$(MOZ_APP_VERSION) $(DIST)/bin $(srcdir) $(XPI_MODULE) $(DLL_SUFFIX) $(LIB_PREFIX)
 
 clean::
 	test -f $(DIST)/bin/$(XPIFILE) && rm $(DIST)/bin/$(XPIFILE) || echo ""
